src/Admin/OptionsMenu.php

```diff
@@ -0,0 +1,490 @@
+
+	/**
+	 * GoogleFonts instance.
+	 *
+	 * @var GoogleFonts
+	 */
+	private $google_fonts;
+
+	/**
+	 * ReaderThemes instance.
+	 *
+	 * @var ReaderThemes
+	 */
+	private $reader_themes;
+
+	/**
+	 * Check whether the conditional object is currently needed.
+	 *
+	 * @return bool Whether the conditional object is needed.
+	 */
+	public static function is_needed() {
+		if ( ! is_admin() ) {
+			return false;
+		}
+
+		/**
+		 * Filter whether to enable the AMP settings.
+		 *
+		 * @since 0.5
+		 * @param bool $enable Whether to enable the AMP settings. Default true.
+		 */
+		$short_circuit = apply_filters( 'amp_options_menu_is_enabled', true );
+
+		if ( true !== $short_circuit ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * OptionsMenu constructor.
+	 *
+	 * @param GoogleFonts  $google_fonts An instance of the GoogleFonts service.
+	 * @param ReaderThemes $reader_themes An instance of the ReaderThemes class.
+	 */
+	public function __construct( GoogleFonts $google_fonts, ReaderThemes $reader_themes ) {
+		$this->google_fonts  = $google_fonts;
+		$this->reader_themes = $reader_themes;
+	}
+
+	/**
+	 * Adds hooks.
+	 */
+	public function register() {
+		add_action( 'admin_post_amp_analytics_options', 'AMP_Options_Manager::handle_analytics_submit' );
+		add_action( 'admin_menu', [ $this, 'add_menu_items' ], 9 );
+
+		$plugin_file = preg_replace( '#.+/(?=.+?/.+?)#', '', AMP__FILE__ );
+		add_filter( "plugin_action_links_{$plugin_file}", [ $this, 'add_plugin_action_links' ] );
+
+		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_assets' ] );
+	}
+
+	/**
+	 * Add plugin action links.
+	 *
+	 * @param array $links Links.
+	 * @return array Modified links.
+	 */
+	public function add_plugin_action_links( $links ) {
+		return array_merge(
+			[
+				'settings' => sprintf(
+					'<a href="%1$s">%2$s</a>',
+					esc_url( add_query_arg( 'page', AMP_Options_Manager::OPTION_NAME, admin_url( 'admin.php' ) ) ),
+					__( 'Settings', 'amp' )
+				),
+			],
+			$links
+		);
+	}
+
+	/**
+	 * Add menu.
+	 */
+	public function add_menu_items() {
+		/*
+		 * Note that the admin items for Validated URLs and Validation Errors will also be placed under this admin menu
+		 * page when the current user can manage_options.
+		 */
+		add_menu_page(
+			__( 'AMP Options', 'amp' ),
+			__( 'AMP', 'amp' ),
+			'manage_options',
+			AMP_Options_Manager::OPTION_NAME,
+			[ $this, 'render_screen' ],
+			self::ICON_BASE64_SVG
+		);
+
+		add_submenu_page(
+			AMP_Options_Manager::OPTION_NAME,
+			__( 'AMP Settings', 'amp' ),
+			__( 'General', 'amp' ),
+			'manage_options',
+			AMP_Options_Manager::OPTION_NAME
+		);
+
+		add_settings_section(
+			'general',
+			false,
+			'__return_false',
+			AMP_Options_Manager::OPTION_NAME
+		);
+
+		add_settings_section(
+			'validation',
+			false,
+			'__return_false',
+			AMP_Options_Manager::OPTION_NAME
+		);
+
+		add_settings_field(
+			Option::SUPPORTED_TEMPLATES,
+			__( 'Supported Templates', 'amp' ),
```

Comment:
For good measure, all `__()` functions should be replaced with `esc_html__()` in this method.

Reply:
Done 6f0a9c84b